File: apps/android/app/src/main/java/com/questionpro/model/WebPage.java
package com.questionpro.model;

import com.alibaba.fastjson.JSONObject;

/**
 * Created by sachinsable on 18/08/16.
 */

public class WebPage {
    public long id;
    public String title;
    public MainMenu mainMenu;
    public ContextMenu contextMenu;
    public int memberCount = -1;
    public ContextMenuContent contextMenuContent;
    // @minu Show dynamic and static image on Navigation for Performance Review.
    public String image = "";
    public boolean isStatic;
    // @minu for Objective and Goals.
    public boolean showMenu;
    public boolean showStat;
    public boolean showCloseButton;

    public static WebPage fromJSON(JSONObject jsonObject){
        WebPage webPage = new WebPage();
        if(jsonObject.containsKey("title")){
            webPage.title = jsonObject.getString("title");
        }
        if (jsonObject.containsKey("image")) {
            webPage.image = jsonObject.getString("image");
        }

        // @minu Set variable for the key showStat
        if (jsonObject.containsKey("showStat")) {
            webPage.showStat = (jsonObject.getBoolean("showStat"));
        }

        //@minu Set variable for the key isStatic
        if (jsonObject.containsKey("isStatic")) {
            webPage.isStatic = jsonObject.getBoolean("isStatic");
        }

        //@minu Set variable for the key showMenu
        if (jsonObject.containsKey("showMenu")) {
            webPage.showMenu = jsonObject.getBoolean("showMenu");
        }

        if (jsonObject.containsKey("contextMenu")) {
            webPage.contextMenu = ContextMenu.fromJSON(jsonObject.getJSONObject("contextMenu"));
        } else {
            webPage.contextMenu = new ContextMenu();
        }

        //@minu set variable for the key contextMenuContent
        if (jsonObject.containsKey("contextMenuContent")) {
            webPage.contextMenuContent = ContextMenuContent.fromJSON(jsonObject.getJSONObject("contextMenuContent"));
        } else {
            webPage.contextMenuContent = new ContextMenuContent();
        }
        if(jsonObject.containsKey("body")){
            JSONObject body = jsonObject.getJSONObject("body");
            if(body.containsKey("memberCount")){
                webPage.memberCount = body.getInteger("memberCount");
            }
        }


        //@sachin for CX
        if(jsonObject.containsKey("showCloseButton")){
            webPage.showCloseButton = jsonObject.getBoolean("showCloseButton");
        }

        return  webPage;
    }
}

File: apps/ios/ios/GeneralUtilityModule/Model/WebPage.swift
    //
//  WebPage.swift
//  QuestionProCommunities
//
//  Created by Jignesh Raiyani on 8/29/16.
//  Copyright © 2016 QuestionPro Pvt Ltd. All rights reserved.
//

import UIKit

class WebPage: NSObject {

    var id : NSNumber?
    var title : String?
    var memberCount : NSNumber?
    var mainMenu =  [MainMenu]()
    var contextMenu =  [ContextMenu]()
    
    // @sujan Show employee image or static done button on the navigation bar.
    var rightButtonImg: String?
    var isStatic = false
    
    // @sujan For portions of objectives and goals.
    var showMenuIcon = false
    var showStat = true
    // @sachin CX close button
    var showCloseButton = false
    override init() {
    }
    
    class func fromJSON (responseDict : [String: Any]) -> WebPage {
        
        let webPage = WebPage()
        if let title = responseDict["title"] as? String {
            webPage.title = title
        }
        if let body = responseDict["body"] as? [String: Any],
            let memberCount = body["memberCount"] as? Int {
            GlobalData.writeEmployDetailsToPlist(memberCount: String(memberCount))
        }
        if let contextMenu = responseDict["contextMenu"] as? [String:Any] {
            webPage.contextMenu = [ContextMenu.fromJSON(menuDict: contextMenu)]
        }
        // @sujan Show employee image or static done button on the navigation bar.
        if let image = responseDict["image"] as? String {
            webPage.rightButtonImg = image
        }
        if let isStatic = responseDict["isStatic"] as? Bool {
            webPage.isStatic = isStatic
        }
        if let showMenu = responseDict["showMenu"] as? Bool {
            webPage.showMenuIcon = showMenu
        }
        if let showStat = responseDict["showStat"] as? Bool {
            webPage.showStat = showStat
        }
        if let showCloseButton = responseDict["showCloseButton"] as? Bool {
            webPage.showCloseButton = showCloseButton
        }
        return webPage
    }

    
}
